packages/web-api/src/methods.ts

```diff
@@ -2103,16 +2103,34 @@ export interface UsersProfileSetArguments extends WebAPICallOptions, TokenOverri
   value?: string; // usable if `profile` is not passed
 }
 
-export interface ViewsOpenArguments extends WebAPICallOptions, TokenOverridable {
+interface ViewsOpenTriggerRequired extends WebAPICallOptions, TokenOverridable {
   trigger_id: string;
+  interactivity_pointer?: never;
   view: View;
 }
 
-export interface ViewsPushArguments extends WebAPICallOptions, TokenOverridable {
+interface ViewsOpenInteractivityRequired extends WebAPICallOptions, TokenOverridable {
+  interactivity_pointer: string;
+  trigger_id?: never;
+  view: View;
+}
+
+export type ViewsOpenArguments = ViewsOpenTriggerRequired | ViewsOpenInteractivityRequired;
+
+interface ViewsPushTriggerRequired extends WebAPICallOptions, TokenOverridable {
   trigger_id: string;
+  interactivity_pointer?: never;
   view: View;
 }
 
+interface ViewsPushInteractivityRequired extends WebAPICallOptions, TokenOverridable {
+  interactivity_pointer: string;
+  trigger_id?: never;
+  view: View;
+}
+
+export type ViewsPushArguments = ViewsPushTriggerRequired | ViewsPushInteractivityRequired;
```

Comment:
Union type working very nicely 🚀 

Can we add some comments here to leave context on why either trigger_id or interactivity_pointer is required to help maintainers in the future?